src/lib/message/L1Transaction.ts

```diff
@@ -184,13 +198,7 @@ export class L1TransactionReceipt implements TransactionReceipt {
     l2SignerOrProvider: T,
     messageIndex?: number
   ): Promise<IL1ToL2MessageReader | IL1ToL2MessageWriter> {
-    return (await isNitroL2(l2SignerOrProvider))
-      ? L1ToL2Message.fromNitro(
-          (await this.nitroReceipt.getL1ToL2Messages(l2SignerOrProvider))[
-            messageIndex || 0
-          ]
-        )
-      : this.classicReceipt.getL1ToL2Message(l2SignerOrProvider, messageIndex)
+    return (await this.getL1ToL2Messages(l2SignerOrProvider))[messageIndex || 0]
```

Comment:
This function is differ from the spec
>   * If the messageIndex is supplied the message at that index will be returned.
>   * If no messageIndex is supplied a message will be returned if this transaction only created one message
>   * All other cases throw an error

It return `undefined` if the message is not found instead of throw

Reply:
It also don''t throw when no messageIndex is supplied and the transaction created more than 1 message

Reply:
Updated